frontend/src/js/controllers/challengeCtrl.js

```diff
@@ -674,6 +675,11 @@
                     for (var i = 0; i < details.results.length; i++) {
                         vm.submissionVisibility[details.results[i].id] = details.results[i].is_public;
                     }
+                    
+                    for (var i = 0; i < details.results.length; i++) {
```

Comment:
Why are we using another loop? I think we can do it in the same loop as written in line `675-677`

Reply:
Thanks for pointing it out!